lib/async.js

```diff
@@ -0,0 +1,17 @@
+const RiffError = require('./riff-error');
+
+module.exports.guardWithTimeout = async (hookFn, timeoutInMs) => {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+        const timerId = setTimeout(() => {
+            reject(new RiffError('error-hook-timeout', 'The hook took too long to run. Aborting now'));
+        }, timeoutInMs);
+
+        try {
+            await hookFn();
```

Comment:
maybe the function will return something we care about. We should also clear the timeout after the promise resolves

```suggestion
            Promise.resolve(hookFn())
                .then(resolve)
                .finally(() => clearTimeout(timerId)));
```

Reply:
slightly adapted to handle errors:
```
const handleError = (reject) => (err) => {
    return reject(new RiffError('error-hook-runtime-error', err));
};

module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
    return new Promise((resolve, reject) => {
        const timerId = setTimeout(() => {
            reject(new RiffError('error-hook-timeout', 'The hook took too long to run. Aborting now'));
        }, timeoutInMs);

        try {
            Promise.resolve(hookFn())
                .then(resolve, handleError(reject))
                .finally(() => clearTimeout(timerId));
        } catch (err) {
            handleError(reject)(err);
        }
    });
};
```